### DISCUSSION TITLE
Feature Request: Pushdown filename filters where possible

### DISCUSSION THREAD

**Discussion by tehunter** [duckdb/duckdb#10178] [Ideas]
I believe that duckdb is currently scanning (loading) all of the matching files, and then filtering to only the selected file, in the following query:
```sql
with source as (
    select * from read_csv_auto("path/to/files/*.txt", filename=true)
)
select * from source where filename=(select max(filename) from source)
```

I'm not 100% sure it's loading every file, but based on the time to run the query it seems like it must be doing that.

Is it possible to optimize the filename subquery? Since `max(...)` only depends on the distinct list of filenames, the query doesn't actually need to load any of the data in the files to compute the value of `max(filename)`.

A related issue is #5706, which would provide some validation into whether or not the other files are being scanned.

**Comment by soerenwolfers** [comment]
You might alternatively be able to use parquet files and store whatever data is encoded in the filename in a (constant) column. Parquet metadata might then get you close to the efficiency you're after (and the boost alone from reading parquets instead of csvs might be enough to solve your performance problems).
